### PR TITLE
HSEARCH-2456 Wait for index status does not raise an exception

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-2456

It appears we were not checking that the index at least existed in some situations, and more importantly we were not waiting for the index status to reach the expected level in some situations. This PR fixes that.

### CODE REVIEW

**Comment by Sanne** [comment]
great, merged!

**Comment by Sanne** [comment]
I had only one comment - which can be done later - about the logging message org.hibernate.search.elasticsearch.logging.impl.Log.unexpectedIndexStatus(String, String, String)

do you think we should rephrase it to point out that we waited some n milliseconds before failing?

**Comment by yrodiere** [comment]
@Sanne Sure: #1260
